submission/programs/adae.R

```diff
@@ -0,0 +1,283 @@
+  #---------------------------------------------------------------------#
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = vars(USUBJID, AEBODSYS, AEDECOD),
+      order = vars(ASTDT, AESEQ),
+      new_var = AOCCPFL,
+      mode = "first"
+    ), filter = TRTEMFL == "Y"
+  ) %>%
+  #---------------------------------------------------------------------#
+  # AOCC02FL - 1st Occurrence 02 Flag for Serious                       #
+  #---------------------------------------------------------------------#
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = vars(USUBJID),
+      order = vars(ASTDT, AESEQ),
+      new_var = AOCC02FL,
+      mode = "first"
+    ), filter = TRTEMFL == "Y" & AESER == "Y"
+  ) %>%
+  #---------------------------------------------------------------------#
+  # AOCC03FL - 1st Occurrence 03 Flag for Serious SOC                   #
+  #---------------------------------------------------------------------#
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = vars(USUBJID, AEBODSYS),
+      order = vars(ASTDT, AESEQ),
+      new_var = AOCC03FL,
+      mode = "first"
+    ), filter = TRTEMFL == "Y" & AESER == "Y"
+  ) %>%
+  #---------------------------------------------------------------------#
+  # AOCC04FL - 1st Occurrence 04 Flag for Serious PT                    #
+  #---------------------------------------------------------------------#
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = vars(USUBJID, AEBODSYS, AEDECOD),
+      order = vars(ASTDT, AESEQ),
+      new_var = AOCC04FL,
+      mode = "first"
+    ), filter = TRTEMFL == "Y" & AESER == "Y"
+  ) %>%
+  # --------------------------------------------------------------------#
+  # CQ01NAM - Customized Query 01 Name                                  #
+  # --------------------------------------------------------------------#
+  mutate(CQ01NAM = ifelse(str_detect(AEDECOD, "APPLICATION") |
+    str_detect(AEDECOD, "DERMATITIS") |
+    str_detect(AEDECOD, "ERYTHEMA") |
+    str_detect(AEDECOD, "BLISTER") |
+    str_detect(AEBODSYS, "SKIN AND SUBCUTANEOUS TISSUE DISORDERS") &
+      !str_detect(AEDECOD, "COLD SWEAT|HYPERHIDROSIS|ALOPECIA"),
+  "DERMATOLOGIC EVENTS",
+  NA_character_
+  )) %>%
+  # --------------------------------------------------------------------#
+  # AOCC01FL - 1st Occurrence 01 Flag for CQ01                          #
+  # --------------------------------------------------------------------#
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = vars(USUBJID),
+      order = vars(ASTDT, AESEQ),
+      new_var = AOCC01FL,
+      mode = "first"
+    ), filter = TRTEMFL == "Y" & CQ01NAM == "DERMATOLOGIC EVENTS"
+  )
+
+#---------------------- #
```

Comment:
Empty section

Reply:
Are you suggesting I remove the comments from lines 45-47 & 247-249?

Reply:
There is a 'section' mentioned here but with no code inside

Reply:
These coments are just to note the START and END of the ADAE derivation programming.  I think this can be better layed out when I simplify the comment structure.  I see the comments from the other scripts and will follow a similar approach.  Will let you know when ready.

Reply:
commenting layout has been updated, please pull and review once more.  If ok, please `Resolve conversation`.

Reply:
@SHAESEN2 or @thomas-neitmann If ok, please `Resolve conversation`.